### PR TITLE
LibWeb: XHR cleanups

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp [hunk 1]
```diff
@@ -626,6 +626,30 @@ WebIDL::ExceptionOr<void> XMLHttpRequest::set_timeout(u32 timeout)
...
+    // 1. If this’s state is not unsent or opened, then throw an "InvalidStateError" DOMException.
+    if (m_state != State::Unsent && m_state != State::Opened)
+        return WebIDL::InvalidStateError::create(realm, "XHR readyState is not UNSENT or OPENED");
+
+    // 2. If this’s send() flag is set, then throw an "InvalidStateError" DOMException.
+    if (m_send)
+        return WebIDL::InvalidStateError::create(realm, "XHR send() flag is already set");
+
+    // 3. Set this’s cross-origin credentials to the given value.
+    m_cross_origin_credentials = with_credentials;
```
Comment (davidot) [inline Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp:650]: Looks like you forgot a `return {}` here

File: Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h [hunk 2]
```diff
@@ -63,8 +63,11 @@ class XMLHttpRequest final : public XMLHttpRequestEventTarget {
 
     WebIDL::ExceptionOr<void> override_mime_type(String const& mime);
 
-    WebIDL::ExceptionOr<void> set_timeout(u32 timeout);
     u32 timeout() const;
+    WebIDL::ExceptionOr<void> set_timeout(u32 timeout);
```
Comment (davidot) [inline Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h:67]: Should moving this be a separate commit?
(I don't think it's needed but you do have a commit fixing some stuff in this file already)

**Review by davidot — CHANGES_REQUESTED** [review]
Two small comments, one of which to fix the error the CI is giving

File: Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp:650]: oops, I wonder how that compiled locally... :thinking:

File: Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibWeb/XHR/XMLHttpRequest.h:67]: Don't think that's worth its own commit tbh, it's only to keep them consistent when adding the other getter/setter pair.

File: Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp (hunk 1 above)
Comment (davidot) [reply Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp:650]: Different warnings enabled?
Maybe because of compiling via ladybird?

Comment (linusg) [reply Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp:648]: No sane compiler should let that through with any set of options. it's literally "function returns something but you didn't return anything" :upside_down_face:

**Review by kennethmyhra — APPROVED** [review]
LGTM!
